src/client/utils/url-resolver.js

```diff
@@ -0,0 +1,59 @@
+import nativeMethods from '../sandbox/native-methods';
+import { ensureTrailingSlash } from '../../utils/url';
+import { get } from '../utils/destination-location';
+
+const DOCUMENT_URL_RESOLVER = 'hammerhead|document-url-resolver';
+
+function createResolver (doc) {
+    var htmlDocument = doc.implementation.createHTMLDocument('title');
+    var a            = nativeMethods.createElement.call(htmlDocument, 'a');
+    var base         = nativeMethods.createElement.call(htmlDocument, 'base');
+
+    nativeMethods.appendChild.call(htmlDocument.body, a);
+    nativeMethods.appendChild.call(htmlDocument.head, base);
+
+    return htmlDocument;
+}
+
+function getResolver (doc) {
+    // NOTE: IE cleans the document up after document.open is called.
+    if (!doc[DOCUMENT_URL_RESOLVER])
+        doc[DOCUMENT_URL_RESOLVER] = createResolver(doc);
+
+    return doc[DOCUMENT_URL_RESOLVER];
```

Comment:
I forgot, why we need to always get it from the document? Why we can't store it in the var?